kratos/solving_strategies/strategies/explicit_solving_strategy.h

```diff
@@ -0,0 +1,657 @@
+     * @return True if the mesh is moved, false otherwise
+     */
+    bool MoveMeshFlag()
+    {
+        return mMoveMeshFlag;
+    }
+
+    /**
+     * @brief This function is designed to move the mesh
+     * @note Be careful it just consider displacements, derive this method to adapt to your own strategies (ALE, FSI, etc...)
+     */
+    virtual void MoveMesh()
+    {
+        KRATOS_TRY
+
+        auto& r_nodes_array = GetModelPart().Nodes();
+        block_for_each(
+            r_nodes_array,
+            [](Node<3>& rNode){
+                noalias(rNode.Coordinates()) = rNode.GetInitialPosition().Coordinates();
+                noalias(rNode.Coordinates()) += rNode.FastGetSolutionStepValue(DISPLACEMENT);
+            }
+        );
+
+        KRATOS_INFO_IF("ExplicitSolvingStrategy", this->GetEchoLevel() > 0) << "Mesh moved." << std::endl;
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Operations to get the pointer to the model
+     * @return mrModelPart: The model part member variable
+     */
+    inline ModelPart& GetModelPart()
+    {
+        return mrModelPart;
+    };
+
+    /**
+     * @brief Operations to get the pointer to the explicit builder and solver
+     * @return mpExplicitBuilder: The explicit builder and solver
+     */
+    inline ExplicitBuilderPointerType& pGetExplicitBuilder()
+    {
+        return mpExplicitBuilder;
+    };
+
+    /**
+     * @brief Operations to get the residual norm
+     * @return The residual norm
+     */
+    virtual double GetResidualNorm()
+    {
+        // Get the required data from the explicit builder and solver
+        const auto p_explicit_bs = pGetExplicitBuilder();
+        auto& r_dof_set = p_explicit_bs->GetDofSet();
+
+        // Calculate the explicit residual
+        p_explicit_bs->BuildRHS(GetModelPart());
+
+        // Calculate the residual norm
+        double res_norm = 0.0;
+        res_norm = block_for_each<SumReduction<double>>(
+            r_dof_set,
+            [](DofType &rDof){return rDof.GetSolutionStepReactionValue();}
+        );
+
+        return res_norm;
+    }
+
+    /**
+     * @brief Function to perform expensive checks.
+     * @details It is designed to be called ONCE to verify that the input is correct.
+     */
+    virtual int Check()
+    {
+        KRATOS_TRY
+
+        // Check if displacement var is needed
+        if (mMoveMeshFlag == true) {
+            VariableUtils().CheckVariableExists<>(DISPLACEMENT, GetModelPart().Nodes());
+        }
+
+        // Elements check
+        for (const auto& r_elem :  GetModelPart().Elements()) {
+            const auto& r_process_info = GetModelPart().GetProcessInfo();
+            r_elem.Check(r_process_info);
+        }
+
+        // Conditions check
+        for (const auto& r_cond :  GetModelPart().Conditions()) {
+            const auto& r_process_info = GetModelPart().GetProcessInfo();
+            r_cond.Check(r_process_info);
+        }
+
```

Comment:
can you add the check for constraints?

Reply:
OK. But I don't think the constraints are compatible yet.